cmd/minikube/cmd/config/util_test.go

```diff
@@ -111,3 +112,26 @@ func TestIsAddonAlreadySet(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateProfile(t *testing.T) {
+	testCases := []struct {
+		profileName string
+		expectErr   string
+	}{
+		{
+			profileName: "not_so_common",
+			expectErr:   "Profile you are trying to switch is not found",
+		},
+	}
+
+	for _, test := range testCases {
+		profileNam := test.profileName
+		expectedMsg := test.expectErr
+
+		err := ValidateProfile(profileNam)
+		fmt.Println(err.Error())
+		if err.Error() != expectedMsg {
```

Comment:
While it's OK to assert that an error satisfies a property such as including the name of the profile, error strings should not be compared directly: it's too fragile. See https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully for more details. 

There are a couple of ways to do this, but here's my recommendation:

- Give this error a type in util.go. For instance:

```go
type ErrProfileDoesNotExist struct {
  Name string
}

func (e ErrProfileDoesNotExist) Error() string {
	return fmt.Sprintf("profile does not exist: %q", e.Name)
}
```

- Update `ValidateProfile` to return `ErrProfileDoesNotExist{Name: profile}`
- Update this test to check the error against this type:

```
switch err := errors.Cause(err).(type) {
  case ErrProfileDoesNotExist:
    return nil
  case default:
    t.Errorf("ValidateProfile error: %v, want: %v")
```

This way you get rich error messages and the ability to compare errors. Hope this helps!